packages/cli/cloud/src/config/local.ts

```diff
@@ -0,0 +1,59 @@
+import path from 'path';
+import os from 'os';
+import fs from 'fs';
+import XDGAppPaths from 'xdg-app-paths';
+
+const APP_FOLDER_NAME = 'com.strapi.cli';
+
+export const CONFIG_FILENAME = 'config.json';
+
+export type LocalConfig = {
+  token?: string;
+};
+
+function checkDirectoryExists(directoryPath: string) {
+  try {
+    return fs.lstatSync(directoryPath).isDirectory();
+  } catch (e) {
+    return false;
+  }
+}
+
+// Determine storage path based on the operating system
+export function getTmpStoragePath() {
+  const storagePath = path.join(os.tmpdir(), APP_FOLDER_NAME);
+  if (!checkDirectoryExists(storagePath)) {
+    fs.mkdirSync(storagePath, { recursive: true });
+  }
+  return storagePath;
+}
+
+function getConfigPath() {
+  const configDirs = XDGAppPaths(APP_FOLDER_NAME).configDirs();
```

Comment:
After the mvp I think it would be great to switch to os keychain support with something like [keytar](https://www.npmjs.com/package/keytar).

But for this, a user on a system can only be logged in to one strapi cloud project. Could we append the strapi cloud project name or something so that different logins can be used for different projects the user might be working with?

Reply:
I'm unsure about the keychain support, as I think it would ask for the user's permission. By using XDGAppPaths we are mimicking Vercel for example.

Reply:
It would allow the user control the security level on the token and keeps it encrypted.

For example, I think most systems would allow users to configure their keychain to ask for verification before the token is accessed (so, someone sitting down at their system couldn't deploy without re-entering password/fingerprint/whatever). But I think what we're doing here is still secure enough for most cases.

I do think a separation by project would be extremely helpful though, otherwise someone with multiple clients on Strapi cloud would have to log in and out between each project, or accidentally attempt to deploy a project with the wrong account (though I assume it wouldn't work).

Alternatively, a login manager to switch between them, but that's a huge feature in itself 😆

Reply:
> side note, Also this CLI could be used to automate deployment so we need to support tokens via the CLI in the CI.

What about `strapi deploy --config=custom/path/to/file` ? Or support for an env var somewhere?

Reply:
Yeah not sure I see the point for that extra layer of security as it's short lived tokens & we could ask for the user to relogin to the cloud or sth like that if we really wanted to.